src/lightning_lite/utilities/device_parser.py

```diff
@@ -0,0 +1,316 @@
+import multiprocessing
+import os
+from typing import Any, List, MutableSequence, Optional, Tuple, Union
+
+import torch
+
+# TODO(lite): Fix the imports
+# from lightning_lite.plugins.environments import TorchElasticEnvironment
+# from lightning_lite.strategies.launchers.multiprocessing import _is_forking_disabled
+from lightning_lite.utilities.exceptions import MisconfigurationException
+from lightning_lite.utilities.types import _DEVICE
+
+
+def determine_root_gpu_device(gpus: List[_DEVICE]) -> Optional[_DEVICE]:
+    """
+    Args:
+        gpus: non-empty list of ints representing which gpus to use
+
+    Returns:
+        designated root GPU device id
+
+    Raises:
+        TypeError:
+            If ``gpus`` is not a list
+        AssertionError:
+            If GPU list is empty
+    """
+    if gpus is None:
+        return None
+
+    if not isinstance(gpus, list):
+        raise TypeError("gpus should be a list")
```

Comment:
```suggestion
        raise TypeError("GPUs should be a list")
```

Reply:
Here it refers specifically to the argument named "gpus", so I don't think it applies here :)